test/tests/materials/discrete/recompute_boundary_error.i

```diff
@@ -5,74 +5,72 @@
     nx = 10
     ny = 1
   []
-  [./left_domain]
+  [left_domain]
     input = gen
     type = SubdomainBoundingBoxGenerator
     bottom_left = '0 0 0'
     top_right = '0.5 1 0'
     block_id = 10
-  [../]
+  []
 []
 
-
 [Variables]
-  [./u]
+  [u]
     initial_condition = 2
-  [../]
+  []
 []
 
 [Kernels]
-  [./diff]
-    type = MatDiffusionTest
+  [diff]
+    type = Diffusion
     variable = u
-    prop_name = 'p'
```

Comment:
can you explain? I'm assuming coupling the recompute material doesn't work anymore (and it shouldn't). But this test does not seem to use `f` at all. Is it really testing the NewtonMaterial. Is the test testing an error message?

Reply:
Yes, this test is testing an error message. The expected error message is something like
```
*** ERROR ***
Incompatible material and object boundaries:
    /home/gary/projects/moose/test/tests/materials/discrete/recompute_boundary_error.i:49: (Materials/recompute_props/boundary) material defined on boundaries 3, 
    /home/gary/projects/moose/test/tests/materials/discrete/recompute_boundary_error.i:59: (Materials/newton/boundary) object needs material on boundaries 1, 3, 
```
Because the `NewtonMaterial` is defined on boundaries `left` and `right`, while the `RecomputeMaterial` is only defined on `left`.